generate_new_lesions/augment_lesions_intra_dataset.py

```diff
@@ -0,0 +1,495 @@
+                             "(e.g., basel-mp2rage)")
+    parser.add_argument("-path-out", type=str, required=True,
+                        help="Path where the augmented dataset will be saved.")
+    parser.add_argument("-num", default=100, type=int, help="Total number of newly generated subjects. Default: 100")
+    parser.add_argument("-seed", default=99, type=int, help="Random seed used for subject mixing. Default: 99")
+    parser.add_argument("-resample", default=False, action='store_true',
+                        help="Resample the augmented images to the resolution of pathological dataset. Default: False")
+    parser.add_argument("-use-n-healthy", default=1, type=int,
+                        help="Number of healthy controls to use for augmentation Default: num_patho // 2")
+    parser.add_argument('--fraction_patho_data', default=1.0, type=float,
+                    help='Fraction of pathology data to be used for training. '
+                    'Use 0.5 for half, 0.25 for 1/4th, etc. ') 
+    parser.add_argument('-split', nargs='+', required=False, type=float, default=[0.6, 0.2, 0.2],
+                        help='Ratios of training, validation and test splits lying between 0-1. Example: --split 0.6 0.2 0.2')
+    parser.add_argument("-qc", default=False, action='store_true', 
+                        help="Perform QC using sct_qc. QC will be saved under -path-out. Default: False")
+    # parser.add_argument("-histogram", default=False, action='store_true', help="Create histograms. Default: False")
+    # parser.add_argument("-min-lesion-vol", "--min-lesion-volume", default=200, type=float,
+    #                     help="Minimum lesion volume in mm^3. Default: 200")
+
+    return parser
+
+
+def insert_lesion(im_augmented, im_augmented_lesion, im_patho_data, im_patho_sc_dil_data, patho_lesion_data, 
+                  im_healthy_sc_data, coords, new_position, lesion_sc_ratio_patho):
+    """"
+    Insert lesion from the bounding box to the im_augmented
+    """
+    # Get bounding box coordinates
+    x0, y0, z0 = coords.min(axis=0)
+    x1, y1, z1 = coords.max(axis=0) + 1  # slices are exclusive at the top
+
+    # Get coordinates where to insert the lesion
+    x, y, z = new_position
+
+    # Compute the shift to the center of the bounding box
+    shift_x, shift_y, shift_z = (x1 - x0) // 2, (y1 - y0) // 2, (z1 - z0) // 2
+
+    # The new origin is shifted to the center of the bounding box (unlike the bottom left earlier)
+    for x_step, x_cor in zip(range(-shift_x, shift_x + 1), range(x0, x1)):
+        for y_step, y_cor in zip(range(-shift_y, shift_y + 1 ), range(y0, y1)):
+            for z_step, z_cor in zip(range(-shift_z, shift_z + 1), range(z0, z1)):
+
+                # make sure that the new lesion is not projected outside of the SC
+                if patho_lesion_data[x_cor, y_cor, z_cor] > 0 and im_healthy_sc_data[x + x_step, y + y_step, z + z_step] > 0:
```

Comment:
suggest to rename `im_healthy_sc_data` into `im_healthy_sc_mask`

Reply:
Done in https://github.com/ivadomed/model_seg_sci/pull/45/commits/e2ac381f9ed44ecdad0b696273186082d920b604